### PR TITLE
DEV: implement discourse-plugin-skeleton template

### PR DESCRIPTION
**Pull request by vinothkannans**

### CODE REVIEW

File: .github/workflows/plugin-linting.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        run: |
+          yarn prettier -v
+          if [ 0 -lt $(find assets -type f \( -name "*.scss" -or -name "*.js" -or -name "*.es6" \) 2> /dev/null | wc -l) ]; then
+            yarn prettier --list-different "assets/**/*.{scss,js,es6}"
+          fi
+          if [ 0 -lt $(find test -type f \( -name "*.js" -or -name "*.es6" \) 2> /dev/null | wc -l) ]; then
+            yarn prettier --list-different "test/**/*.{js,es6}"
+          fi
+
+      - name: Rubocop
```
Comment (CvX) [inline .github/workflows/plugin-linting.yml:51]: This is the version I've been recently including in theme component workflows:

```suggestion

      - name: Ember template lint
        if: ${{ always() }}
        run: |
          if [ 0 -lt $(find assets/javascripts -type f -name "*.hbs" 2> /dev/null | wc -l) ]; then
            yarn ember-template-lint assets/javascripts
          fi

      - name: Rubocop
```

File: assets/javascripts/discourse/initializers/extend-for-salesforce.js.es6 [hunk 2]
```diff
@@ -59,6 +61,7 @@ function initializeWithApi(api) {
 
   if (isStaff) {
     api.modifyClass("raw-view:topic-status", {
+      // eslint-disable-next-line no-undef
```
Comment (CvX) [inline assets/javascripts/discourse/initializers/extend-for-salesforce.js.es6:64]: Is ember global usage going to be sorted out in another PR? 😃

File: assets/javascripts/discourse/initializers/extend-for-salesforce.js.es6 [hunk 3]
```diff
@@ -76,6 +79,7 @@ function initializeWithApi(api) {
     });
 
     TopicStatusIcons.addObject(["has_salesforce_case", "briefcase", "case"]);
+    // eslint-disable-next-line no-undef
```
Comment (CvX) [inline assets/javascripts/discourse/initializers/extend-for-salesforce.js.es6:82]: And Discourse global usage? 😉

File: package.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "discourse-salesforce",
+  "version": "0.1.0",
+  "repository": "git@github.com:discourse/discourse-salesforce.git",
+  "author": "Vinoth Kannan",
+  "license": "MIT",
+  "devDependencies": {
+    "@arkweid/lefthook": "^0.7.2",
+    "eslint-config-discourse": "^2.0.0"
```
Comment (CvX) [inline package.json:9]:
```suggestion
    "eslint-config-discourse": "^3.0.0"
```

**Review by CvX — APPROVED** [review]

File: assets/javascripts/discourse/initializers/extend-for-salesforce.js.es6 (hunk 2 above)
Comment (vinothkannans) [reply assets/javascripts/discourse/initializers/extend-for-salesforce.js.es6:64]: I think we don't have a solution for this. We're doing the same in core too
https://github.com/discourse/discourse/blob/bf33d2cd4b27bbb63720c8691f4508e4ea0e8568/app/assets/javascripts/discourse/app/models/site.js#L218

File: assets/javascripts/discourse/initializers/extend-for-salesforce.js.es6 (hunk 3 above)
Comment (vinothkannans) [reply assets/javascripts/discourse/initializers/extend-for-salesforce.js.es6:82]: I think we don't have a solution for this. We're doing the same in core too
https://github.com/discourse/discourse/blob/bf33d2cd4b27bbb63720c8691f4508e4ea0e8568/app/assets/javascripts/discourse/app/models/site.js#L218
